ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl.hs

```diff
@@ -205,7 +206,10 @@ openDBInternal args launchBgTasks = do
         testing = Internal
           { intCopyToImmutableDB       = getEnv  h Background.copyToImmutableDB
           , intGarbageCollect          = getEnv1 h Background.garbageCollect
-          , intUpdateLedgerSnapshots   = getEnv  h Background.updateLedgerSnapshots
+          , intUpdateLedgerSnapshots   = getEnv  h \case
```

Comment:
I have not spent much time getting familiar with this initialisation. I did what seemed reasonable but maybe there are some dragons here that I'm not aware of.

Reply:
Looks like this testing interface is only used in the `test-storage` test suite. The QSM testing machinery there can generate a "take a snapshot" instruction as part of its test case. So I think you're implementation here seems correct (ie equivalent to what it was before).

If only to stick to the general pattern in this `openDBInternal` function, I suggest defining this lambda alongside `updateLedgerSnapshots`.

In fact, maybe the name `updateLedgerSnapshots` should _still_ be this function you've written here, ie that variable name should keep its former semantics. So we'd move this to `Background.hs` and rename what is currently there. I'm not sure what to call the new function (the one that is given the snapshot to write)... maybe `updateLedgerSnapshotsWith` `addLedgerSnapshot`?

(This comment got big, so let's chat about it on Tuesday.)

Reply:
We decided on our call to just pull it out in a local binding; without doing any of the renaming I suggested above.